integration-tests/soak/vrfv2_test.go

```diff
@@ -0,0 +1,363 @@
+		os := &client.VRFV2TxPipelineSpec{
+			Address: coordinator.Address(),
+		}
+		ost, err := os.String()
+		require.NoError(t, err)
+		nativeTokenKeyAddress, err = chainlinkNode.PrimaryEthAddress()
+		require.NoError(t, err)
+		_, err = chainlinkNode.MustCreateJob(&client.VRFV2JobSpec{
+			Name:                     fmt.Sprintf("vrf-%s", jobUUID),
+			CoordinatorAddress:       coordinator.Address(),
+			FromAddresses:            []string{nativeTokenKeyAddress},
+			EVMChainID:               fmt.Sprint(chainClient.GetNetworkConfig().ChainID),
+			MinIncomingConfirmations: minimumConfirmations,
+			PublicKey:                pubKeyCompressed,
+			ExternalJobID:            jobUUID.String(),
+			ObservationSource:        ost,
+			BatchFulfillmentEnabled:  false,
+		})
+		require.NoError(t, err)
+		provingKey, err := actions.EncodeOnChainVRFProvingKey(*vrfKey)
+		require.NoError(t, err)
+		err = coordinator.RegisterProvingKey(
+			nativeTokenKeyAddress,
+			provingKey,
+		)
+		require.NoError(t, err)
+		encodedProvingKeys = append(encodedProvingKeys, provingKey)
+	}
+
+	keyHash, err := coordinator.HashOfKey(context.Background(), encodedProvingKeys[0])
+	require.NoError(t, err)
+
+	evmKeySpecificConfigTemplate := `
+[[EVM.KeySpecific]]
+Key = '%s'
+
+[EVM.KeySpecific.GasEstimator]
+PriceMax = '%d gwei'
+`
+	//todo - make evmKeySpecificConfigTemplate for multiple eth keys
+	evmKeySpecificConfig := fmt.Sprintf(evmKeySpecificConfigTemplate, nativeTokenKeyAddress, maxGasPriceGWei)
+	tomlConfigWithUpdates := fmt.Sprintf("%s\n%s", config.BaseVRFV2NetworkDetailTomlConfig, evmKeySpecificConfig)
+
+	newEnvLabel := "updatedWithRollout=true"
+	newTestEnvironment := setupVRFV2Environment(t, testNetwork, tomlConfigWithUpdates, testEnvironment.Cfg.Namespace, newEnvLabel)
+
+	err = newTestEnvironment.RolloutStatefulSets()
+	require.NoError(t, err, "Error performing rollout restart for test environment")
+
+	conds := &env_client.ReadyCheckData{
+		ReadinessProbeCheckSelector: newEnvLabel,
+		Timeout:                     5 * time.Minute,
+	}
+
+	err = newTestEnvironment.RunCustomReadyConditions(conds, 0)
+
+	require.NoError(t, err, "Error running test environment")
+
+	//need to get node's urls again since port changed after redeployment
+	chainlinkNodes, err = client.ConnectChainlinkNodes(newTestEnvironment)
+	require.NoError(t, err)
+
+	vrfV2SoakTest := testsetups.NewVRFV2SoakTest(&testsetups.VRFV2SoakTestInputs{
+		BlockchainClient:     chainClient,
+		TestDuration:         testInputs.TestDuration,
+		ChainlinkNodeFunding: testInputs.ChainlinkNodeFunding,
+		SubscriptionFunding:  testInputs.SubscriptionFunding,
+		StopTestOnError:      testInputs.StopTestOnError,
+		RequestsPerMinute:    testInputs.RequestsPerMinute,
+		TestFunc: func(t *testsetups.VRFV2SoakTest, requestNumber int, wg *sync.WaitGroup) error {
+
+			concurrentEVMClient, err := blockchain.ConcurrentEVMClient(testNetwork, newTestEnvironment, chainClient)
+			consumer.ChangeEVMClient(concurrentEVMClient)
```

Comment:
Doesn't look like a safe usage of `consumer`. This can change the client value in consumer unpredictably from multiple go routine.